src/Microsoft.Extensions.ML/ModelReloadToken.cs

```diff
@@ -44,6 +44,12 @@ public ModelReloadToken()
         /// <summary>
         /// Used to trigger the change token when a reload occurs.
         /// </summary>
-        public void OnReload() => _cts.Cancel();
+        public void OnReload()
+        {
+            if (!_cts.IsCancellationRequested)
+                _cts.Cancel();
+            else
+                Console.WriteLine("ModelReloadToken has already been Canceled.");
+        }
```

Comment:
ML.NET is a library that can be used in environments where there is no console. I don't think we should be using Console.WriteLine here. 
What is the desired behavior? Is this for information only to the user or is this an error condition?

Reply:
This is just to output some info into test command, this is not for user but rather a error info, I just output some info to see whether this is common test crash case.

---
In reply to: [350532971](https://github.com/dotnet/machinelearning/pull/4503#discussion_r350532971) [](ancestors = 350532971)

Reply:
We should not be writing to the console inside of our libraries.